src/openzeppelin/tests/token/erc20/test_dual20.cairo

```diff
@@ -0,0 +1,364 @@
+use openzeppelin::token::erc20::ERC20;
+use openzeppelin::token::erc20::interface::IERC20;
+use openzeppelin::token::erc20::interface::IERC20Camel;
+use openzeppelin::token::erc20::interface::IERC20Dispatcher;
+use openzeppelin::token::erc20::interface::IERC20CamelDispatcher;
+use openzeppelin::token::erc20::interface::IERC20DispatcherTrait;
+use openzeppelin::token::erc20::interface::IERC20CamelDispatcherTrait;
+use openzeppelin::token::erc20::dual20::DualERC20Trait;
+use openzeppelin::token::erc20::dual20::DualERC20;
+use openzeppelin::tests::mocks::snake20_mock::SnakeERC20Mock;
+use openzeppelin::tests::mocks::camel20_mock::CamelERC20Mock;
+use openzeppelin::tests::mocks::erc20_panic::SnakeERC20Panic;
+use openzeppelin::tests::mocks::erc20_panic::CamelERC20Panic;
+use openzeppelin::tests::mocks::non721_mock::NonERC721;
+use openzeppelin::tests::utils;
+
+use openzeppelin::tests::utils::PanicTrait;
+
+///
+/// Constants
+///
+
+const NAME: felt252 = 111;
+const SYMBOL: felt252 = 222;
+const DECIMALS: u8 = 18_u8;
+
+fn SUPPLY() -> u256 {
+    2000_u256
+}
+fn VALUE() -> u256 {
+    300_u256
+}
+fn OWNER() -> ContractAddress {
+    contract_address_const::<10>()
+}
+fn SPENDER() -> ContractAddress {
+    contract_address_const::<20>()
+}
+fn RECIPIENT() -> ContractAddress {
+    contract_address_const::<30>()
+}
+fn OPERATOR() -> ContractAddress {
+    contract_address_const::<40>()
+}
+
+///
+/// Setup
+///
+
+fn setup_snake() -> (DualERC20, IERC20Dispatcher) {
+    let mut calldata = ArrayTrait::new();
+    calldata.append(NAME);
+    calldata.append(SYMBOL);
+    calldata.append(SUPPLY().low.into());
+    calldata.append(SUPPLY().high.into());
+    calldata.append(OWNER().into());
+    let target = utils::deploy(SnakeERC20Mock::TEST_CLASS_HASH, calldata);
+    (DualERC20 { target: target }, IERC20Dispatcher { contract_address: target })
+}
+
+fn setup_camel() -> (DualERC20, IERC20CamelDispatcher) {
+    let mut calldata = ArrayTrait::new();
+    calldata.append(NAME);
+    calldata.append(SYMBOL);
+    calldata.append(SUPPLY().low.into());
+    calldata.append(SUPPLY().high.into());
+    calldata.append(OWNER().into());
+    let target = utils::deploy(CamelERC20Mock::TEST_CLASS_HASH, calldata);
+    (DualERC20 { target: target }, IERC20CamelDispatcher { contract_address: target })
+}
+
+fn setup_non_erc20() -> DualERC20 {
+    let calldata = ArrayTrait::new();
+    let target = utils::deploy(NonERC721::TEST_CLASS_HASH, calldata);
+    DualERC20 { target: target }
+}
+
+fn setup_erc20_panic() -> (DualERC20, DualERC20) {
+    let snake_target = utils::deploy(SnakeERC20Panic::TEST_CLASS_HASH, ArrayTrait::new());
+    let camel_target = utils::deploy(CamelERC20Panic::TEST_CLASS_HASH, ArrayTrait::new());
+    (DualERC20 { target: snake_target }, DualERC20 { target: camel_target })
+}
+
+///
+/// Case agnostic methods
+///
+
+#[test]
+#[available_gas(2000000)]
+fn test_dual_name() {
+    let (snake_dispatcher, _) = setup_snake();
```

Comment:
```suggestion
    let (snake_dual_dispatcher, _) = setup_snake();
```

Sinece both outputs are technically dispatchers, maybe adding this for the dual one make things easier to read?

Reply:
Hmm you are correct. My take is that `dispatcher` is referring to __the__ (dual) dispatcher. To your point, maybe it's not as clear as it could be...I'll apply the suggestion and see what we think

Reply:
Resolved offline